### PR TITLE
Fix network plugin config option load from collection

### LINKED ISSUE #63975: Can't use `vars` in playbook when using collections
##### SUMMARY

When using collection with HTTP API plugin, it's not possible to use any kind of `vars` from the playbook
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

ansible collection

##### ANSIBLE VERSION

```paste below
2.9
```

##### CONFIGURATION

```paste below

```

##### STEPS TO REPRODUCE

Following play fails when trying to fetch `vars` value through `get_option('domain')` (ref: https://github.com/ansible/ansible/pull/63976/files#diff-911386a804691dadae5f292b64df16a9R47) when used via `ansible collections` but works as expected and gets the `vars` value when used directly from `core` side.

```yaml
- hosts: checkpoint
  connection: httpapi
  tasks:
    - name: create new mgmt
      cp_mgmt_host:
        name: "Sales Web Server 2"
        ip_address: 10.0.10.121
      register: show_network_response
      vars:
        ansible_checkpoint_domain: "System Data"
```

##### EXPECTED RESULTS

Get the `vars` value from ansible collections

##### ACTUAL RESULTS

Fail to retrieve the `vars` value from ansible collections side

```paste below

```

### PR DESCRIPTION
**Pull request by ganeshrn**
##### SUMMARY

Fixes #63975

* Update the complete sub-plugin name within network
  connection plugins to handle sub-plugin in collection
  scenario.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

plugins/connection/httpapi.py
plugins/connection/netconf.py
plugins/connection/network_cli.py
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/plugins/connection/httpapi.py [hunk 1]
```diff
@@ -202,7 +202,8 @@ def __init__(self, play_context, new_stdin, *args, **kwargs):
             self.httpapi = httpapi_loader.get(self._network_os, self)
```
Comment (justjais) [inline lib/ansible/plugins/connection/httpapi.py:202]: @ganeshrn ideally the `httpapi_loader` should also be called from fqcn

File: lib/ansible/plugins/connection/netconf.py [hunk 2]
```diff
@@ -229,7 +229,8 @@ def __init__(self, play_context, new_stdin, *args, **kwargs):
         netconf = netconf_loader.get(self._network_os, self)
```
Comment (justjais) [inline lib/ansible/plugins/connection/netconf.py:229]: same as above

File: lib/ansible/plugins/connection/network_cli.py [hunk 3]
```diff
@@ -329,8 +329,9 @@ def __init__(self, play_context, new_stdin, *args, **kwargs):
 
             self.cliconf = cliconf_loader.get(self._network_os, self)
```
Comment (justjais) [inline lib/ansible/plugins/connection/network_cli.py:330]: same as above

**Review by justjais — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/connection/httpapi.py (hunk 1 above)
Comment (ganeshrn) [reply lib/ansible/plugins/connection/httpapi.py:202]: The name of the sub-plugin need to be updated as fqcn which I have done in the latest commit.

**Review by justjais — APPROVED** [review]

**Comment by ganeshrn** [comment]
recheck

**Comment by ganeshrn** [comment]
Failure not related to changes in PR

**Pull request merged**